src/font.rs

```diff
@@ -438,6 +445,69 @@ mod tests {
         assert_eq!(font_obj.features.unwrap(), "# this is the feature from lightWide\n");
     }
 
+    #[test]
+    fn loading_invalid_ufo_dir_path() {
+        let path = "totally/bogus/filepath/font.ufo";
+        let font_load_res = Font::load(path);
+        match font_load_res {
+            Ok(_) => panic!("unxpected Ok result"),
+            Err(Error::MissingUfoDir(_)) => (), // expected value
```

Comment:
`assert_eq!(font_load_res, Err(Error::MissingUfoDir(...)))` or something maybe?

Reply:
Tried.  It won't let me perform equality comparisons on the Error type

Reply:
Two things:

As written, I would do this as,

```rust
assert!(matches!(font_load_res, Err(Error::MissingUfoDir(_))));
```

but there's probably a better option: explicitly testing for a panic with the [`should_panic`](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic) attribute.

Reply:
> but there's probably a better option: explicitly testing for a panic with the should_panic attribute.

Ah nice, I didn't realize that you can add an expected string with `should_panic`.  I tried this approach but I can't elicit a panic from the new custom Error.  The attribute doesn't seem to catch errors like it does panics. I see this in the docs that you linked:

> You can’t use the #[should_panic] annotation on tests that use Result<T, E>. Instead, you should return an Err value directly when the test should fail.

Let me tinker with the test Result approach a bit.  If I can't find a better solution, I'll transition things over the matches! macro approach above.

Reply:
Maybe it would work if you just `unwrap` the `Result`?

Reply:
(but `matches!` is totally reasonable too)

Reply:
It looks like the use of Result in unit tests is good if you expect an Ok() but there isn't a way to expect an Err.  We could return a test error when the Result is Ok 💥 but that seems worse than the matches pattern matching approach :) 

I went with the matches! macro. That works well!

b1c17aba624a771de855f48164edb20defec7959